openwisp_firmware_upgrader/base/models.py

```diff
@@ -761,6 +804,9 @@ def upgrade(self, recoverable=True):
         # meaning the image file is corrupted or not apt for flashing
         except UpgradeAborted:
             self.status = "aborted"
+        # this exception is raised when the upgrade is cancelled by the user
+        except UpgradeCanceled:
```

Comment:
Sorry one thing I noticed only, I may have suggested the alternative spelling, but the status is being spelled as "cancelled", so let's keep consitency and call this:

```suggestion
        except UpgradeCancelled:
```